### PR TITLE
feat: Haskell language

### PR DESCRIPTION
**Pull request by algas**
## Feature

- Haskell language
`echo '{ "name": "David" }' | quicktype -l haskell`

## Test

- pass tests in my local environment
```.sh
npm run build
FEATURE=haskell npm test
```

## Others

- ~~This PR does not support GraphQL in Haskell~~
- GraphQL works with Haskell

### CODE REVIEW

**Comment by schani** [comment]
@algas That's amazing!  Thank you!

What is it that keeps GraphQL from working?  There's nothing specific in the target languages about GraphQL.

**Comment by algas** [comment]
@schani How can I test it for GraphQL files?
These code was written by reference to Elm-language code, but it did not support GraphQL.

**Comment by schani** [comment]
Ah, you'll have to fix a bunch of lint warning.  You can locally run

`npm run tslint`

I'm actually not sure about GraphQL and Elm now.  It might be because it only supports a single top-level?  I can look into it if you'd like.

**Comment by schani** [comment]
I haven't tried it out, but GraphQL should work with Elm.  Please check out [our blog post about GraphQL](https://blog.quicktype.io/graphql-with-quicktype/).

**Comment by algas** [comment]
Thank you for your information about GraphQL.
I fixed lint error and added configuration for ci tests.
GraphQL works with Haskell now in some cases.

I will fix some tests failed.

**Comment by schani** [comment]
Fantastic, thank you!

File: test/fixtures/haskell/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/elm-stuff
```
Comment (schani) [inline test/fixtures/haskell/.gitignore:1]: I'm guessing this is not needed?

Comment (algas) [reply test/fixtures/haskell/.gitignore:1]: Thank you

**Comment by algas** [comment]
@schani What can I do to fix "buildkite/quicktype-pr/diff"?

**Comment by schani** [comment]
I'll see if I have time to fix the diff issue on the weekend.  It's a quicktype build bug, not your bug.

**Comment by schani** [comment]
Thank you so much for your contribution! 🙏

**Pull request merged**